user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleDetailsComponent.java

```diff
@@ -219,12 +218,9 @@ private void updateSampleGridDataProvider(ExperimentId experimentId, String filt
       sortOrders.add(SortOrder.of("sampleCode").ascending());
       return sampleInformationService.queryPreview(experimentId, query.getOffset(),
           query.getLimit(), List.copyOf(sortOrders), filter).stream();
-    }, query -> SampleDetailsComponent.this.sampleInformationService.countPreviews(
-        experimentId,
-        filter));
-    sampleGrid.getLazyDataView()
-        .addItemCountChangeListener(
-            countChangeEvent -> experimentTab.setSampleCount(countChangeEvent.getItemCount()));
+    });
+    sampleGrid.getLazyDataView().addItemCountChangeListener(
```

Comment:
NIT: You are accessing the sample grid while it is modified. Consider assigning a variable rather than directly modifying the field.